cw_bitcoin/lib/litecoin_wallet.dart

```diff
@@ -178,13 +219,544 @@ abstract class LitecoinWalletBase extends ElectrumWallet with Store {
+      try {
+        await updateAllUnspents();
+        await updateTransactions();
+        await updateBalance();
+        syncStatus = SyncedSyncStatus();
+      } catch (e, s) {
+        print(e);
+        print(s);
+        syncStatus = FailedSyncStatus();
+      }
+      return;
+    }
+
+    await waitForMwebAddresses();
+    try {
+      await getStub();
+      await processMwebUtxos();
+      await updateTransactions();
+      await updateUnspent();
+      await updateBalance();
+    } catch (e) {
+      print("failed to start mweb sync: $e");
+      syncStatus = FailedSyncStatus();
+      return;
+    }
+
+    _syncTimer?.cancel();
+    _syncTimer = Timer.periodic(const Duration(milliseconds: 1500), (timer) async {
+      if (syncStatus is FailedSyncStatus) return;
+
+      final nodeHeight =
+          await electrumClient.getCurrentBlockChainTip() ?? 0; // current block height of our node
+      final resp = await CwMweb.status(StatusRequest());
+
+      if (resp.blockHeaderHeight < nodeHeight) {
+        int h = resp.blockHeaderHeight;
+        syncStatus = SyncingSyncStatus(nodeHeight - h, h / nodeHeight);
+      } else if (resp.mwebHeaderHeight < nodeHeight) {
+        int h = resp.mwebHeaderHeight;
+        syncStatus = SyncingSyncStatus(nodeHeight - h, h / nodeHeight);
+      } else if (resp.mwebUtxosHeight < nodeHeight) {
+        syncStatus = SyncingSyncStatus(1, 0.999);
+      } else {
+        // prevent unnecessary reaction triggers:
+        if (syncStatus is! SyncedSyncStatus) {
+          syncStatus = SyncedSyncStatus();
+        }
+
+        if (resp.mwebUtxosHeight > walletInfo.restoreHeight) {
+          await walletInfo.updateRestoreHeight(resp.mwebUtxosHeight);
+          await checkMwebUtxosSpent();
+          // update the confirmations for each transaction:
+          for (final transaction in transactionHistory.transactions.values) {
+            if (transaction.isPending) continue;
+            int txHeight = transaction.height ?? resp.mwebUtxosHeight;
+            final confirmations = (resp.mwebUtxosHeight - txHeight) + 1;
+            if (transaction.confirmations == confirmations) continue;
+            transaction.confirmations = confirmations;
+            transactionHistory.addOne(transaction);
+          }
+          await transactionHistory.save();
+        }
+        return;
+      }
+    });
+
+    // setup a watch dog to restart the sync process if it gets stuck:
+    List<double> lastFewProgresses = [];
+    _stuckSyncTimer?.cancel();
+    _stuckSyncTimer = Timer.periodic(const Duration(seconds: 10), (timer) async {
+      if (syncStatus is! SyncingSyncStatus) return;
+      if (syncStatus.progress() > 0.98) return; // don't check if we're close to synced
+      lastFewProgresses.add(syncStatus.progress());
+      if (lastFewProgresses.length < 10) return;
+      // limit list size to 10:
+      while (lastFewProgresses.length > 10) {
+        lastFewProgresses.removeAt(0);
+      }
```

Comment:
it's intentionally long as it was triggering unnecessarily, but it's more meant so that background sync doesn't get stuck forever more than anything

we can probably just move this code into the background sync branch so that it applies to all wallet types, since it doesn't do much for the foreground state anyways

Reply:
removed in the latest commit (and added to the mweb-bg-sync-2 branch)